src/pymatgen/io/vasp/inputs.py

```diff
@@ -905,72 +965,15 @@ def from_str(cls, string: str) -> Self:
                     params[key] = cls.proc_val(key, val)
         return cls(params)
 
-    @staticmethod
-    def proc_val(key: str, val: str) -> list | bool | float | int | str:
+    @classmethod
+    def proc_val(cls, key: str, val: str) -> list | bool | float | int | str:
```

Comment:
Have to move these recordings (list_keys/bool_keys/...) as class variable for `check_params` to access them, also make `proc_val` class method. 

I don't think this would be breaking but still good for others to double check